### DISCUSSION TITLE
Error fulltextsearch

### DISCUSSION THREAD

**Discussion by 4y55** [nextcloud/all-in-one#4873] [Questions] [answered]
It worked, it stopped after some update, I don't know when because I didn't notice.

Nextcloud AIO v9.0.1
[Nextcloud Hub 8](https://next.ayslanalves.com/) (29.0.2)
Full text search - Elasticsearch Platform 29.0.1
Full text search - Files 29.0.0

dockercompose:
`version: "3.8"

volumes:
  nextcloud_aio_mastercontainer:
    name: nextcloud_aio_mastercontainer # This line is not allowed to be changed

services:
  nextcloud:
    image: nextcloud/all-in-one:latest # Must be changed to 'nextcloud/all-in-one:latest-arm64' when used with an arm64 CPU
    restart: always
    container_name: nextcloud-aio-mastercontainer # This line is not allowed to be changed
    volumes:
      - nextcloud_aio_mastercontainer:/mnt/docker-aio-config # This line is not allowed to be changed
      - /var/run/docker.sock:/var/run/docker.sock:ro # May be changed on macOS, Windows or docker rootless. See the applicable documentation
    ports:
      - 8080:8080
      - 8443:8443 # Can be removed when running behind a reverse proxy. See https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md
    environment: # Is needed when using any of the options below
      - APACHE_PORT=11000 # Is needed when running behind a reverse proxy. See https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md
      # - APACHE_IP_BINDING=127.0.0.1 # Should be set when running behind a reverse proxy that is running on the same host. See https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md
      # - TALK_PORT=3478 # This allows to adjust the port that the talk container is using.
      - NEXTCLOUD_DATADIR=/mnt/user/arquivos/nextcloud-aio # Allows to set the host directory for Nextcloud's datadir. See https://github.com/nextcloud/all-in-one#how-to-change-the-default-location-of-nextclouds-datadir
      # - NEXTCLOUD_MOUNT=/mnt/ # Allows the Nextcloud container to access the chosen directory on the host. See https://github.com/nextcloud/all-in-one#how-to-allow-the-nextcloud-container-to-access-directories-on-the-host
      # - DOCKER_SOCKET_PATH=/var/run/docker.sock # Needs to be specified if the docker socket on the host is not located in the default '/var/run/docker.sock'. Otherwise mastercontainer updates will fail.
      - DISABLE_BACKUP_SECTION=false # Setting this to true allows to hide the backup section in the AIO interface.
      # - NEXTCLOUD_UPLOAD_LIMIT=10G # Can be adjusted if you need more. See https://github.com/nextcloud/all-in-one#how-to-adjust-the-upload-limit-for-nextcloud
      # - NEXTCLOUD_MAX_TIME=3600 # Can be adjusted if you need more. See https://github.com/nextcloud/all-in-one#how-to-adjust-the-max-execution-time-for-nextcloud
      - NEXTCLOUD_MEMORY_LIMIT=2048M # Can be adjusted if you need more. See https://github.com/nextcloud/all-in-one#how-to-adjust-the-php-memory-limit-for-nextcloud`

`docker exec --user www-data nextcloud-aio-nextcloud php occ fulltextsearch:test
 
.Testing your current setup:  
Creating mocked content provider. ok  
Testing mocked provider: get indexable documents. (2 items) ok  
Loading search platform. (Elasticsearch) ok  
Testing search platform. ok  
Locking process ok  
Removing test. ok  
Pausing 3 seconds 1 2 3 ok  
Initializing index mapping. ok  
Indexing generated documents. ok  
Pausing 3 seconds 1 2 3 ok  
Retreiving content from a big index (license). (size: 32386) ok  
Comparing document with source. ok  
Searching basic keywords:  
 - 'test' (result: 0, expected: ["simple"]) fail  
Error detected, unlocking process 
ok In Test.php line 676:
                                                                               
  Unexpected SearchResult: {"provider":{"id":"test_provider","name":"Test Pro  
  vider"},"platform":{"id":"elastic_search","name":"Elasticsearch"},"document  
  s":[],"info":[],"meta":{"timedOut":false,"time":9,"count":0,"total":0,"maxS  
  core":0}}                                                                    
                                                                               

fulltextsearch:test [--output [OUTPUT]] [-j|--json] [-d|--platform_delay PLATFORM_DELAY]`

`docker exec --user www-data nextcloud-aio-nextcloud php occ fulltextsearch:reset
WARNING! You are about to reset your indexed documents:
- provider: ALL
- collection: ALL

Do you really want to reset your indexed documents ? (y/N) 
aborted.`

**Comment by szaimen** [comment, marked as answer]
Hi, can you follow https://github.com/nextcloud/all-in-one/discussions/2509 and check if that makes it work again?

**Comment by 4y55** [comment]
Partially resolved, thank you!

Now I can test with OK return, and I can index, but it gives an error.

Initially it gave an error in a PDF document of less than 16mb, I checked the document and everything was ok, no errors, no passwords.

I removed the document and it gave an error in another one. Do you know what could cause this?

The nextcloud logs also show an error.

```
docker exec --user www-data nextcloud-aio-nextcloud php occ fulltextsearch:index
Options: []
Options: []
Options: []
Memory: 44 MB
┌─ Indexing  ────
│ Action: indexDocument
│ Provider: Files                Account: admin
│ Document: 118807
│ Info: application/pdf
│ Title: 01-Livros/Direito Processual Penal/Direito Processual Penal - Renato Brasileiro 2020.pdf
│ Content size: 19235132
│ Chunk:     18/127
│ Progress:      7/8
└──
┌─ Results ────
│ Result:      0/0
│ Index: 
│ Status: 
│ Message: 
│ 
│ 
└──
┌─ Errors ────
│ Error:     28/28
│ Index: files:21882
│ Exception: OCA\FullTextSearch_Elasticsearch\Vendor\Elastic\Elasticsearch\Exception\ClientResponseException
│ Message: unknown error
│ 
│ 
└──
## x:first result ## c/v:prec/next result ## b:last result
## f:first error ## h/j:prec/next error ## d:delete error ## l:last error
## q:quit ## p:pause 

In ElasticSearchPlatform.php line 202:
                                                              
  [OCA\FullTextSearch\Exceptions\PlatformTemporaryException]  
                                                              
fulltextsearch:index [--output [OUTPUT]] [-r|--no-readline] [--] [<options>]
```

![erro](https://github.com/nextcloud/all-in-one/assets/50604378/901af088-eec3-4d28-bf94-6123174cc23b)
